sqlglot/dialects/spark.py

```diff
@@ -18,12 +14,6 @@ def _create_sql(self, e):
     return create_with_partitions_sql(self, e)
 
 
-def _map_sql(self, expression):
```

Comment:
why did you remove this?

Reply:
So when I started added in the map functions from the PySpark DataFrame API I noticed that we weren't actually using the map function from Spark but rather creating maps using `MAP_FROM_ARRAYS` even if there weren't arrays. Since Spark had direct support for creating maps like one would expect I removed this workaround.

Reply:
map in spark is Map(a, b, c, d)
in order to do arrays you need to do MAP_FROM_ARRAYS, are you sure this is correct to remove?